system/jlib/jlog.cpp

```diff
@@ -607,33 +675,37 @@ void LogMsg::fprintTable(FILE * handle, unsigned fields) const
 
 void LogMsg::fprintTableHead(FILE * handle, unsigned fields)
 {
-    if(fields & MSGFIELD_audience)
-        fprintf(handle, "Audience ");
-    if(fields & MSGFIELD_class)
-        fprintf(handle, "Class    ");
-    if(fields & MSGFIELD_detail)
-        fprintf(handle, "    Detail ");
-    if(fields & MSGFIELD_msgID)
-        fprintf(handle, "   MsgID ");
-    if(fields & MSGFIELD_date)
-        fprintf(handle, "      Date ");
-    if(fields & MSGFIELD_time)
-        fprintf(handle, "    Time ");
-    if(fields & MSGFIELD_process)
-        fprintf(handle, "  PID ");
-    if(fields & MSGFIELD_thread)
-        fprintf(handle, "  TID ");
-    if(fields & MSGFIELD_session)
-        fprintf(handle, "      SessionID      ");
-    if(fields & MSGFIELD_node)
-        fprintf(handle, "               Node ");
-    if(fields & MSGFIELD_job)
-        fprintf(handle, "  JobID ");
-    if(fields & MSGFIELD_user)
-        fprintf(handle, " UserID ");
-    if(fields & MSGFIELD_component)
-        fprintf(handle, " Compo ");
-    fprintf(handle, "\n\n");
+    StringBuffer  header;
+    loggingFieldColumns.generateHeaderRow(header, fields, true).append("\n");
+    fputs(header.str(), handle);
+}
+
+unsigned getMessageFieldsFromHeader(FILE *handle)
+{
+    unsigned currentFieldHeader = 0;
+    try
+    {
+        MemoryBuffer mb(loggingFieldColumns.getMaxHeaderSize());
+        fpos_t pos;
+        fgetpos (handle,&pos);
+        rewind (handle);
+
+        mb.reserve(loggingFieldColumns.getMaxHeaderSize());
+        const char * line = fgets (static_cast<char *>(mb.bufferBase()),  loggingFieldColumns.getMaxHeaderSize(), handle );
+        currentFieldHeader = loggingFieldColumns.extractMessageFieldsFromHeader(line, true);
```

Comment:
fgets could return null (if file is empty).  It should not be called if that is the case, but safer to protect against it.